src/appleseed/foundation/image/analysis.cpp

```diff
@@ -157,28 +145,23 @@ double compute_rms_deviation(const Image& image1, const Image& image2)
 
     double mse = 0.0;   // mean square error
 
-    for (size_t ty = 0; ty < props.m_tile_count_y; ++ty)
+    for (size_t y = 0; y < props.m_canvas_height; ++y)
     {
-        for (size_t tx = 0; tx < props.m_tile_count_x; ++tx)
+        for (size_t x = 0; x < props.m_canvas_width; ++x)
         {
-            const Tile& tile1 = image1.tile(tx, ty);
-            const size_t tile_width = tile1.get_width();
-            const size_t tile_height = tile1.get_height();
+            float image_color[4];
+            image1.get_pixel(x, y, image_color);
```

Comment:
You can do the same with Color4f:

    Color4f color1;
    image1.get_pixel(x, y, color1);

Also, if you call the arguments of this function `image1` and `image2`, there's no such thing as a "ref image", just call the colors `color1` and `color2`.

Reply:
I opted to use the component route so that we could relax the constraint that channel counts must match to perform this analysis since the error computation is rgb specific.

For instance, this seems to be the method used by https://github.com/appleseedhq/appleseed/blob/master/src/appleseed/foundation/math/sampling/imageimportancesampler.h#L308 to support RGB/RGBA.

(Maybe we don't care to support RGB in this case, and enforcing RGBA is okay..)

Reply:
I see, that's a good point.